uproot/tree.py

```diff
@@ -286,13 +286,19 @@ def _attachstreamer(self, branch, streamer, streamerinfosmap, isTClonesArray):
 
                 self._attachstreamer(subbranch, submembers.get(name, None), streamerinfosmap, isTClonesArray)
 
+    def _addprovenance(self, branch, context):
+        parents = [context.treename]
+        for x in branch.itervalues(recursive=True):
+            x._provenance = parents + [branch.name]
```

Comment:
It's a one-time change now, which is good, but doesn't this need to recursively pass down `parents` (different in each level of subbranch)? Do you have any tests that are deeper than one level—because that's what it looks like would go wrong here.

Reply:
Good point.
So `tree/branch/subbranch/subsubbranch` should have 
```python
subsubbranch._provenance = [tree.name, branch.name, subbranch.name]
```
I do not have an example, but should be easy enough to construct one.
Are there any candidates in the existing test samples?

Reply:
I don't remember any in the hand-made samples, but the set of user-provided samples (all named `issue*.root`) could have anything. You need multiple levels of struct nesting without variable-length list nesting.

Reply:
Have an example now and as expected the results are wrong (missing intermediate branches)
```
E             Full diff:
E             - [b'b1', b'tree/b1/b2', b'tree/b1/b3', b'tree/b1/b4']
E             + [b'b1', b'tree/b1/b2', b'tree/b1/b2/b3', b'tree/b1/b2/b3/b4']
```

unit test created, now trying to fix it